hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMCertStore.java

```diff
@@ -56,20 +61,21 @@
 /**
  * A Certificate Store class that persists certificates issued by SCM CA.
  */
-public class SCMCertStore implements CertificateStore {
+public final class SCMCertStore implements CertificateStore {
   private static final Logger LOG =
       LoggerFactory.getLogger(SCMCertStore.class);
   private SCMMetadataStore scmMetadataStore;
   private final Lock lock;
   private AtomicLong crlSequenceId;
 
-  public SCMCertStore(SCMMetadataStore dbStore, long sequenceId) {
+  private SCMCertStore(SCMMetadataStore dbStore, long sequenceId) {
     this.scmMetadataStore = dbStore;
     lock = new ReentrantLock();
     crlSequenceId = new AtomicLong(sequenceId);
   }
 
   @Override
+  @Replicate
```

Comment:
You'd better place `@Replicate` inside the interface `CertificateStore`. Please check `ContainerStateManagerV2` and `ContainerStateManagerImpl` as an example.

Reply:
Replicate class in hadoop-hdds-server-scm and CertificateStore in hdds-server-framework. 
hadoop-hdds-server-scm depends on framework. So, taken a simple approach and placed annotation in SCMCertSttore. 
So, to move @Replication to CertificateStore I need to move the Replicate Class to hdds-server-framework. Let me do it.

Reply:
I just run `TestSCMCertStore` locally, and not seen the trace 
```
    LOG.info("Invoking method {} on target {}, cost {}us",
        method, ratisHandler, (Time.monotonicNowNanos() - startTime) / 1000.0);
```
seems the annotation does not work if placing in subclass.

@bshashikant Above trace is sufficient to indicate whether annotated method goes through ratis, even in unit test.

Reply:
Its not preferrable solution i feel, to check for specific log messages in a test. That's why , i thought replicating over ratis and verifying the db is a better choice.  In this case, i am ok with any of the approaches mentioned.

@GlenGeng , thanks for testing out replicate annotation over the sub class and interface.

Reply:
TestSCMCertStore is not constructing RatisServer that is the reason, we don't see the log.

I have updated TestReplicateAnnotation itself, to call storeAndValidateCertificate.